Editorial review: Document the Summarizer API (#39263)
mdn/content: files/en-us/web/api/summarizer/inputquota/index.md

---
title: "Summarizer: inputQuota property"
short-title: inputQuota
slug: Web/API/Summarizer/inputQuota
page-type: web-api-instance-property
status:
  - experimental
browser-compat: api.Summarizer.inputQuota
---

{{APIRef("Summarizer API")}}{{SeeCompatTable}}

The **`inputQuota`** read-only property of the {{domxref("Summarizer")}} interface returns the input quota available to the browser for generating summaries.

## Value

A number specifying the available input quota. This number is implementation dependant. For example, it might be {{jsxref("infinity")}} if there are no limits beyond the user's memory and the maximum length of JavaScript strings, or it might be a number of tokens in the case of LLMs that use a token/credits scheme.

> EDITORIAL: I'm not sure how this number is useful — the value returned is so subjective that it seems meaningless. I'd like to get some insight into its use cases, and an example of how Chrome implements it.

Author: Cool, this makes sense. I have removed my editorial note, and added an example that shows how to check whether you have enough quota to perform a summary operation.

---
title: "Summarizer: inputQuota property"
short-title: inputQuota
slug: Web/API/Summarizer/inputQuota
page-type: web-api-instance-property
status:
  - experimental
browser-compat: api.Summarizer.inputQuota
---

{{APIRef("Summarizer API")}}{{SeeCompatTable}}

The **`inputQuota`** read-only property of the {{domxref("Summarizer")}} interface returns the input quota available to the browser for generating summaries.

## Value

> A number specifying the available input quota. This number is implementation dependant. For example, it might be {{jsxref("infinity")}} if there are no limits beyond the user's memory and the maximum length of JavaScript strings, or it might be a number of tokens in the case of AI models that use a token/credits scheme.

Reviewer: ```suggestion
A number specifying the available input quota. This number is implementation-dependent. For example, it might be {{jsxref("infinity")}} if there are no limits beyond the user's memory and the maximum length of JavaScript strings, or it might be a number of tokens in the case of AI models that use a token/credits scheme.
```

## Examples

> ### Checking if you have enough quota

Reviewer: Super-nit, but I'm used to seeing "sufficient/insufficient quota", if you agree:

```suggestion
### Checking if you have sufficient quota
```

In the below snippet, we create a new `Summarizer` instance using {{domxref("Summarizer.create_static", "create()")}}, then return the total input quota via `inputQuota` and the input quota usage for a summarizing a particular text string via {{domxref("Summarizer.measureInputUsage", "measureInputUsage()")}}.

We then test to see if the individual input usage for that string is great than the total available quota. If so, we throw an appropriate error; it not, we commence summarizing the string using {{domxref("Summarizer.summarize", "summarize()")}}.

```js
const summarizer = await Summarizer.create({
  sharedContext:
    "A general summary to help a user decide if the text is worth reading",
  type: "tl;dr",
  length: "short",
});

const totalInputQuota = summarizer.inputQuota;
const inputUsage = await summarizer.measureInputUsage(myTextString);

if (inputUsage > totalInputQuota) {

>   throw new Error("Boo, not enough quota left to generate a summary.");
> } else {
>   console.log("Yay, enough quote left to generate a summary.");

Reviewer: If you like it:

```suggestion
  throw new Error("Boo, insufficient quota to generate a summary.");
} else {
  console.log("Yay, quota available to generate a summary.");
```

Reviewer: but check "quote" v "quota"
